django/db/models/functions/comparison.py

```diff
@@ -175,21 +179,28 @@ def join(self, args):
         )
 
     def as_postgresql(self, compiler, connection, **extra_context):
-        if not connection.features.is_postgresql_16:
-            copy = self.copy()
-            copy.set_source_expressions(
-                [
-                    Cast(expression, TextField()) if index % 2 == 0 else expression
-                    for index, expression in enumerate(copy.get_source_expressions())
-                ]
-            )
-            return super(JSONObject, copy).as_sql(
+        copy = self.copy()
+        copy.set_source_expressions(
+            [
+                Cast(expression, TextField()) if index % 2 == 0 else expression
+                for index, expression in enumerate(copy.get_source_expressions())
+            ]
+        )
```

Comment:
I don't like the pattern of using an index and modulus operator to do the transformation here.

I personally would prefer

```python
expressions = copy.get_source_expressions()
expressions[::2] = [Cast(key, TextField()) for key in expressions[::2]]
copy.set_source_expressions(expressions)
```

But that's probably a minor nitpick and certainly not worth including in a PR intended to fix this specific issue.